### DISCUSSION TITLE
Distributed architecture - Forward nodes & agents

### DISCUSSION THREAD

**Discussion by HBA-EMK** [Security-Onion-Solutions/securityonion#8060] [Unsupported Versions]
Hi, 
I am currently testing a distributed S.O. architecture with mainly a manager-search node and few forward nodes.
It took a few tweaks to route syslog data from network devices through filebeat into the forward node, I can see the data within the SOC.
Now I'm trying to setup Wazuh & Winlogbeat the same way : data directed to the forward node and exploited through the manager node.
Having found very few data on the matter, I'm not even sure forward nodes can do that.
I added the client server via SO-allow, and configured the minion SLS file for Wazuh. 

firewall:
     38   assigned_hostgroups:
     39     chain:
     40       DOCKER-USER:
     41         hostgroups:
     45           wazuh_agent:
     46             portgroups:
     47               - portgroups.wazuh_agent

Enrollment fails with the following logs (first through wazuh agent manager, then through agent-auth)  : 

2022/06/01 18:09:23 ossec-agent: INFO: Using notify time: 10 and max time to reconnect: 60
2022/06/01 18:09:23 ossec-agent: INFO: (1410): Reading authentication keys file.
2022/06/01 18:09:23 ossec-agent: INFO: Received exit signal.
2022/06/01 18:12:54 agent-auth: INFO: Starting enrollment process to server: x.x.x.x
2022/06/01 18:13:15 agent-auth: ERROR: Unable to connect to x.x.x.x:1515

I can see that the communication reaches the forward node (tcpdump) but I can't figure out the missing part.

Version : 2.3.120
On-Prem deployment with internet access.
Installed from ISO image
Hardware specs allocated beyond the recommended config.
so-status shows all services as running.

I would appreciate if anyone can help me on that.

Cheers! 

Edit : Added some contextual data as per "Read before posting!"

**Comment by Jaap79** [comment]
What does "dmesg" tell you on the forward node? I did practically the same thing and experienced some issues with IPtables blocking port 1515. SO-Allow seems to resolve a bit, but not everything all the time. After allowing it through iptables on the forward node, it seems to be OK.

I also added the API for autoregistration of the Wazuh agents the same way (port 1514), which makes setting things up a whole lot easier.

**Reply by Jaap79** [reply]
(Oh and if dmesg doesn't say antyhing, check any firewalls that are in between the agents and the fw-node...)
